src/main/java/com/alkemy/wallet/controller/TransactionController.java

```diff
@@ -1,23 +1,26 @@
 package com.alkemy.wallet.controller;
 
-import com.alkemy.wallet.dto.TransactionDetailDto;
+import com.alkemy.wallet.model.Transaction;
+import com.alkemy.wallet.service.TransactionService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.access.prepost.PreAuthorize;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.Optional;
+
 @RestController
 @RequestMapping("/transactions")
 public class TransactionController {
-    //private final TransactionService transactionService;
+    @Autowired
+    TransactionService transactionService;
 
-    /*@GetMapping( value = "/{id}")
+    @GetMapping( value = "/{id}")
     @PreAuthorize("hasRole('USER_ROLE')")
-    public TransactionDetailDto getTransactionDetailById(@PathVariable Integer id ) throws Exception {
+    public Optional<Transaction> getTransactionDetailById(@PathVariable Integer id ) {
```

Comment:
* Change return type to ResponseEntity< Transaction >
* Resolve Optional object in Service Layer